### PR TITLE
Allow blade echo statements in attributes

### PR DESCRIPTION
**Pull request by inxilpro**
Currently, Blade X requires using `:prefixed-attributes` for binding, which doesn't allow for easy refactoring/autocomplete/syntax highlighting in an IDE that can parse Blade (like PhpStorm). This PR adds support for Blade echo statements inside of attributes, such as:

```php
<card title="{{ $title }}">
  My card
</card>
```

And:

```php
<card title="{!! $title_html !!}">
  My card
</card>
```

### CODE REVIEW

**Comment by inxilpro** [comment]
I intentionally left something like `<card title="Hello {{ $name }}" />` out of the scope of this PR, but it's definitely a potential issue—we may at least want to be intentional about throwing an exception or more gracefully handling that use case.

**Comment by inxilpro** [comment]
Actually, I think I have a better solution to that. Hold off on reviewing this for a sec.

**Comment by inxilpro** [comment]
OK, now this should also support concatenation, so something like:

```php
<card title="{{ $name }}'s Progress">
  ...
</card>
```

Will effectively translate to:

```php
@component('components.card', ['title' => e($name).'\'s Progress'])
```

**Comment by inxilpro** [comment]
Alright, one more consideration that I added. Instead of just passing the string concatenation as raw PHP code, like so:

```php
[
  'title' => 'Hello '.e($name),
]
```

I'm now wrapping it in an `HtmlString` object:

```php
[
  'title' => new \Illuminate\Support\HtmlString('Hello '.e($name)),
]
```

This means that if you do something like:

```php
<?php $name = "O'Connor"; ?>
<card title="{{ $name }} Family Tree">
```

And your component looks like:

```php
<div>
  <h3>{{ $title }}</h3>
  {!! $slot !!}
</div>
```

You won't get `<h3>O&amp;#039;Connor Family Tree</h3>` but instead will get `<h3>O&#039;Connor Family tree</h3>`.

**Comment by inxilpro** [comment]
@freekmurze and @AlexVanderbist — I totally get that y'all have more pressing things, so no pressure on the full review. 

It would be helpful for me to know **if this is a feature you're interested in entertaining**, though. I'd like to try to use Blade X in a project, but honestly the ergonomics of the `:colon-prefixed-attributes` just doesn't work for me. If Blade X is just not a good fit, that's fine, but I'd like to know that before going too far down that path if possible.

Thanks!

**Comment by AlexVanderbist** [comment]
Hi @inxilpro, thanks for the PR and sorry to keep you waiting. 

We've thought about this syntax a lot before deciding to go with the `:` prefixed attributes. The main reason we didn't go through with the "blade echo" notation is because it's quite confusing for new users when this will be parsed. In blade context `{{ }}` is usually used for printing variables but in BladeX it would then correspond to passing variables.

For example `<card title="{{ $title }}" />` with `$title='Example'` could mean 3 things:
- blade echo is being parsed first  and the bladex component ends up with `['title' => 'Example']`
- bladex is being parsed first and will pass the string directly to the component: `['title' => '{{ $title }}']`
- bladex will recognize the `{{ $title }}` notation and will know it means to pass the variable (and not print it) resulting in `['title' => $title]`

Even as an alternative (maybe undocumented) syntax I think it would be better to just avoid the confusion altogether. That way, even if someone instinctively tries to pass a variable using the curly brackets, they'll quickly realise that it's just passing the whole thing as a string.

(On top of that we chose for the `:attributes` because that's how Vue does it, a popular choice for front-end components at the time. It's not a great argument against this PR tho).

So tl;dr we're not gonna merge this for now to avoid confusing between the existing syntax, blade echo statements and passing variables. Feel free to maintain a fork for your project with this syntax if you really want to use BladeX in your projects tho :)

Thanks for the PR!
